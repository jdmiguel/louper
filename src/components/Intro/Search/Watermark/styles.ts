import { styled } from '@mui/material/styles';

export const StyledRoot = styled('svg')(({ theme }) => ({
  opacity: 0,
  animation: `${theme.animation.fadeIn} 200ms forwards`,
}));
