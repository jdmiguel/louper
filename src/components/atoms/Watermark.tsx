const Watermark = () => (
  <svg width="326" height="336" viewBox="0 0 326 336">
    <path
      d="M150 300C67.2656 300 0 232.676 0 150C0 67.3242 67.2656 0 150 0C232.734 0 300 67.3242 300 150C300 232.676 232.734 300 150 300ZM150 33.3398C85.6641 33.3398 33.3398 85.6641 33.3398 150C33.3398 214.336 85.6641 266.66 150 266.66C214.336 266.66 266.66 214.336 266.66 150C266.66 85.6641 214.336 33.3398 150 33.3398Z"
      fill="#BBB0B5"
    />
    <path
      d="M229.508 263.622C222.682 256.779 222.696 245.699 229.539 238.873C236.381 232.048 247.462 232.061 254.287 238.904C261.113 245.747 320.448 305.229 320.448 305.229C327.273 312.072 327.259 323.153 320.417 329.978C313.574 336.804 302.494 336.79 295.668 329.947L229.508 263.622Z"
      fill="#BBB0B5"
    />
    <path
      d="M225.059 153.5C225.059 141.137 221.016 130.531 212.988 121.859C213.867 121.449 213.926 117.113 213.047 108.852C212.344 100.59 210.586 92.9727 207.715 86L203.496 86.9961C200.625 87.4062 196.113 89.0469 189.902 91.8594C183.75 94.6719 177.305 98.3633 170.566 102.934C165.996 101.586 159.141 100.941 150 100.941C141.27 100.941 134.414 101.645 129.434 102.934C114.668 92.7383 102.305 87.0547 92.2852 86C89.4141 92.9727 87.6562 100.59 86.9531 108.852C86.1328 117.113 86.1914 121.449 87.0703 121.859C79.043 130.59 75 141.137 75 153.5C75 161.527 75.9375 168.734 77.7539 175.18C79.5703 181.566 81.9141 186.781 84.7266 190.824C87.5977 194.867 91.0547 198.383 95.2148 201.43C99.375 204.477 103.184 206.703 106.641 208.109C110.156 209.574 114.082 210.629 118.594 211.391C123.047 212.211 126.387 212.621 128.73 212.738L136.055 212.914L141.914 213.09L150.059 213.266L158.203 213.09L164.062 212.914L171.387 212.738C173.73 212.562 177.07 212.152 181.523 211.391C186.035 210.629 190.02 209.516 193.477 208.109C196.934 206.703 200.742 204.477 204.902 201.43C209.004 198.383 212.461 194.867 215.332 190.824C218.086 186.781 220.43 181.566 222.305 175.18C224.121 168.734 225.059 161.527 225.059 153.5Z"
      fill="#BBB0B5"
    />
    <path
      d="M191.25 199.57C181.992 203.906 168.398 206.074 150.352 206.074L149.707 206.133C131.66 206.133 118.125 203.965 108.984 199.629C99.8438 195.293 95.2734 186.328 95.2734 172.852C95.2734 164.824 98.0859 158.262 103.711 153.281C106.113 151.113 109.16 149.59 112.852 148.711C116.543 147.832 119.941 147.422 123.105 147.539L136.172 148.359C141.738 148.77 146.191 149.004 150.352 149.004C153.398 149.004 159.023 148.652 167.285 148.008C175.547 147.363 181.699 147.363 185.684 148.008C189.727 148.652 193.301 150.41 196.289 153.223C202.207 158.438 205.137 165 205.137 172.793C205.137 186.328 200.449 195.234 191.25 199.57Z"
      fill="#E0E6EF"
    />
    <path
      d="M183.105 164.277C181.055 161.406 178.652 160 175.781 160C172.793 160 170.215 161.406 168.105 164.277C166.055 167.09 165 170.547 165 174.707C165 178.809 166.055 182.324 168.105 185.137C170.156 187.949 172.734 189.355 175.781 189.355C178.594 189.355 180.996 187.949 183.105 185.137C185.156 182.324 186.211 178.867 186.211 174.707C186.211 170.605 185.156 167.09 183.105 164.277Z"
      fill="#BBB0B5"
    />
    <path
      d="M131.105 165.277C129.055 162.406 126.652 161 123.781 161C120.793 161 118.215 162.406 116.105 165.277C114.055 168.09 113 171.547 113 175.707C113 179.809 114.055 183.324 116.105 186.137C118.156 188.949 120.734 190.355 123.781 190.355C126.594 190.355 128.996 188.949 131.105 186.137C133.156 183.324 134.211 179.867 134.211 175.707C134.211 171.605 133.156 168.09 131.105 165.277Z"
      fill="#BBB0B5"
    />
  </svg>
);

export default Watermark;
