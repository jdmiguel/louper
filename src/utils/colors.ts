export const colors = {
  white: '#e0e6ef',
  pink: '#c94b73',
  lightPink: '#e56e94',
  darkPink: '#711874',
  darkPinkOverlay: 'rgba(71, 19, 104, 0.55)',
  purple: '#571aa7',
  lightPurple: '#6b41dd',
  darkPurple: '#260036',
  grey: '#24282a',
  lightGrey: '#5a5a5a',
  darkGrey: '#998991',
  blue: '#4d40ba',
  lightBlue: '#a5a4e4',
  darkBlue: '#08001A',
  red: '#cf1b55',
  gradient: 'radial-gradient(at 80% 40%, #210b58 58%, #3d0554 100%)',
  darkGradient: 'radial-gradient(at 60% 15%, #08001a 60%, #260036 100%)',
};
