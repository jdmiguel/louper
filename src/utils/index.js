export const errorLiterals = {
  maximumRequest:
    'You have excedeed the maximum allowed request. Please, wait for a while',
  unavailableUser: 'Please, choose an available user',
};

export const externalLink = (url) => window.open(url, '_blank');
