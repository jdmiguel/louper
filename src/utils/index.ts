import { ResponseError } from './types';

enum ErrorMessage {
  MAX = 'You have excedeed the maximum allowed request. Please, wait for a while',
  NO_USER = 'Please, choose an available user',
  GENERIC = 'Sorry! there was an error on the server side.',
}

export const getErrorMessage = (error: ResponseError) => {
  if (error.status === 403) {
    return ErrorMessage.MAX;
  }

  if (error.status === 404) {
    return ErrorMessage.NO_USER;
  }

  return ErrorMessage.GENERIC;
};

export const debounce = (fn: any, delay: number) => {
  let timer: ReturnType<typeof setTimeout>;

  return (...args: any) => {
    if (timer) clearTimeout(timer);
    timer = setTimeout(() => {
      fn(...args);
    }, delay);
  };
};
