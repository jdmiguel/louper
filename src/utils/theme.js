/* material-ui */
import { keyframes } from '@emotion/react';
import { createTheme } from '@mui/material/styles';

export const theme = createTheme({
  overrides: {
    MuiCssBaseline: {
      '@global': {
        html: {
          overflowX: 'hidden',
          background: '#E0E6EF',
        },
        body: {
          margin: 0,
          padding: 0,
        },
        button: {
          boxShadow: 'initial !important',
        },
      },
    },
  },
  typography: {
    fontFamily: ['Comfortaa', 'Arial', 'sans-serif'].join(','),
    htmlFontSize: 12,
    h1: {
      color: '#353D40',
      fontSize: '4.8rem',
      fontWeight: 700,
      letterSpacing: '-0.25rem',
      lineHeight: 0,
      '@media (min-width: 769px)': {
        fontSize: '6rem',
      },
    },
    h2: {
      color: '#353D40',
      fontSize: '3.5rem',
      fontWeight: 300,
      letterSpacing: '-0.1rem',
      lineHeight: 0,
      '@media (min-width: 769px)': {
        fontSize: '4.2rem',
      },
    },
    h3: {
      color: '#353D40',
      fontSize: '1.3rem',
      fontWeight: 700,
      '@media (min-width: 769px)': {
        fontSize: '1.5rem',
      },
    },
    h4: {
      color: '#353D40',
      fontSize: '1.1rem',
      fontWeight: 500,
      '@media (min-width: 769px)': {
        fontSize: '1.2rem',
      },
    },
    h5: {
      color: '#6b77a1',
      fontSize: '0.9rem',
      fontWeight: 700,
      textTransform: 'uppercase',
    },
    h6: {
      color: '#757d81',
      fontSize: '0.8rem',
      fontWeight: 500,
      textTransform: 'uppercase',
    },
    button: {
      fontSize: '0.9rem',
    },
    body1: {
      color: '#353D40',
      fontSize: '0.9rem',
      lineHeight: '1.1rem',
    },
    body2: {
      color: '#E1D7EC',
      fontSize: '0.8rem',
      lineHeight: '1rem',
    },
    subtitle2: {
      color: '#353D40',
      fontSize: '0.8rem',
      fontWeight: 500,
      textTransform: 'uppercase',
    },
  },
  palette: {
    background: {
      default: '#E0E6EF',
    },
    primary: {
      main: '#553285',
      light: '#7B52AB',
      dark: '#36175E',
      contrastText: '#E1D7EC',
    },
    secondary: {
      main: '#6b77a1',
      light: '#c7cde1',
      dark: '#565e7a',
      contrastText: '#bac1d9',
    },
    neutral: {
      main: '#353D40',
    },
  },
  animation: {
    greet: keyframes`
      0%,100% { transform: rotate(0) }
      20%,60% { transform: rotate(-25deg) }
      40%,80% { transform: rotate(10deg) }
    `,
  },
});
