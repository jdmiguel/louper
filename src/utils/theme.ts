/* material-ui */
import { keyframes } from '@emotion/react';
import { createTheme } from '@mui/material/styles';

/* colors */
import { colors } from './colors';

type Animation = {
  greet: any;
  shimmer: any;
};

declare module '@mui/material/styles' {
  interface Theme {
    animation: Animation;
    overrides: any;
  }
  interface ThemeOptions {
    animation: Animation;
    overrides: any;
  }
  interface Palette {
    neutral: Palette['primary'];
    overlay: Palette['primary'];
  }
  interface PaletteOptions {
    neutral: PaletteOptions['primary'];
    overlay: PaletteOptions['primary'];
  }
}

export const theme = createTheme({
  animation: {
    greet: keyframes`
      0%,100% { transform: rotate(0) }
      20%,60% { transform: rotate(-25deg) }
      40%,80% { transform: rotate(10deg) }
    `,
    shimmer: keyframes`
      0% { background-position-x: -1200px}
      100% { background-position-x: 1200px}
    `,
  },
  overrides: [
    {
      MuiCssBaseline: {
        '@global': {
          html: {
            overflowX: 'hidden',
            background: colors.darkBlue,
          },
          body: {
            margin: 0,
            padding: 0,
          },
          button: {
            boxShadow: 'initial !important',
          },
        },
      },
    },
  ],
  typography: {
    fontFamily: ['Comfortaa', 'Arial', 'sans-serif'].join(','),
    htmlFontSize: 12,
    h2: {
      fontSize: '1.1rem',
      fontWeight: 300,
      '@media (min-width: 768px)': {
        fontSize: '1.3rem',
      },
    },
    h3: {
      fontSize: '1.3rem',
      fontWeight: 700,
      '@media (min-width: 768px)': {
        fontSize: '1.5rem',
      },
    },
    h4: {
      fontSize: '1.1rem',
      fontWeight: 500,
      '@media (min-width: 768px)': {
        fontSize: '1.2rem',
      },
    },
    h5: {
      fontSize: '1rem',
      fontWeight: 700,
      textTransform: 'uppercase',
    },
    h6: {
      fontSize: '0.9rem',
      fontWeight: 700,
      textTransform: 'uppercase',
    },
    body1: {
      fontSize: '1em',
    },
    body2: {
      fontSize: '0.9em',
    },
    overline: {
      fontSize: '0.8rem',
      fontWeight: 700,
      lineHeight: '1rem',
      textTransform: 'uppercase',
    },
  },
  palette: {
    primary: {
      main: colors.pink,
      light: colors.lightPink,
      dark: colors.darkPink,
    },
    secondary: {
      main: colors.blue,
      light: colors.lightBlue,
      dark: colors.darkBlue,
    },
    neutral: {
      main: colors.grey,
      light: colors.lightGrey,
    },
    overlay: {
      main: colors.darkPinkOverlay,
    },
    background: {
      default: colors.darkBlue,
    },
    text: {
      primary: colors.white,
      secondary: colors.lightBlue,
    },
  },
});
