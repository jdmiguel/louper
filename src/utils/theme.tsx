import { ReactElement } from 'react';
import { keyframes } from '@emotion/react';
import { Keyframes } from '@emotion/serialize';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { colors } from './colors';

type Animation = {
  greet: Keyframes;
  fadeIn: Keyframes;
  fadeInUp: Keyframes;
  fadeInLeft: Keyframes;
  fadeInRight: Keyframes;
};

declare module '@mui/material/styles' {
  interface Theme {
    animation: Animation;
  }
  interface ThemeOptions {
    animation: Animation;
  }
  interface Palette {
    neutral: Palette['primary'];
    overlay: Palette['primary'];
  }
  interface PaletteOptions {
    neutral: PaletteOptions['primary'];
    overlay: PaletteOptions['primary'];
  }
}

export const theme = createTheme({
  animation: {
    greet: keyframes`
      0%,100% { transform: rotate(0) }
      20%,60% { transform: rotate(-25deg) }
      40%,80% { transform: rotate(10deg) }
    `,
    fadeIn: keyframes`
      0% { opacity:0 }
      100% { opacity:1 }
    `,
    fadeInUp: keyframes`
      0% { opacity:0; transform: translateY(20px); }
      100% { opacity:1; transform: translateY(0px);  }
    `,
    fadeInLeft: keyframes`
      0% { opacity:0; transform: translateX(-50px); }
      100% { opacity:1; transform: translateX(0px);  }
    `,
    fadeInRight: keyframes`
      0% { opacity:0; transform: translateX(50px); }
      100% { opacity:1; transform: translateX(0px);  }
    `,
  },
  components: {
    MuiCssBaseline: {
      styleOverrides: {
        body: {
          overflowX: 'hidden',
        },
        h1: {
          margin: 0,
          lineHeight: 0,
        },
        h2: {
          margin: 0,
          lineHeight: 0,
        },
      },
    },
  },
  typography: {
    fontFamily: ['Comfortaa', 'Arial', 'sans-serif'].join(','),
    htmlFontSize: 12,
    h2: {
      fontSize: '0.9rem',
      '@media (min-width: 480px)': {
        fontSize: '1rem',
      },
      '@media (min-width: 768px)': {
        fontSize: '1.1rem',
      },
      '@media (min-width: 1440px)': {
        fontSize: '1.15rem',
      },
      fontWeight: 300,
    },
    h3: {
      fontSize: '1.3rem',
      fontWeight: 700,
      '@media (min-width: 768px)': {
        fontSize: '1.5rem',
      },
    },
    h4: {
      fontSize: '1.1rem',
      fontWeight: 500,
      '@media (min-width: 768px)': {
        fontSize: '1.2rem',
      },
    },
    h5: {
      fontSize: '1rem',
      fontWeight: 700,
      textTransform: 'uppercase',
    },
    h6: {
      fontSize: '0.9rem',
      fontWeight: 700,
      textTransform: 'uppercase',
    },
    body1: {
      fontSize: '1rem',
    },
    body2: {
      fontSize: '0.9rem',
    },
    overline: {
      fontSize: '0.8rem',
      fontWeight: 700,
      lineHeight: '1rem',
      textTransform: 'uppercase',
    },
  },
  palette: {
    primary: {
      main: colors.pink,
      light: colors.lightPink,
      dark: colors.darkPink,
    },
    secondary: {
      main: colors.blue,
      light: colors.lightBlue,
      dark: colors.darkBlue,
    },
    error: {
      main: colors.red,
    },
    neutral: {
      dark: colors.grey,
      main: colors.lightGrey,
      light: colors.white,
    },
    overlay: {
      main: colors.darkPinkOverlay,
    },
    background: {
      default: colors.darkBlue,
      paper: colors.gradient,
    },
    text: {
      primary: colors.white,
      secondary: colors.lightBlue,
    },
  },
});

export const renderWithTheme = (component: ReactElement) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
