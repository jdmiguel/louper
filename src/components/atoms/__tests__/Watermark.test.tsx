import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { renderWithTheme } from '../../../utils/theme';
import Watermark from '../Watermark';

describe('<Watermark />', () => {
  it('renders the content properly', () => {
    const { container } = render(renderWithTheme(<Watermark />));
    expect(container).toMatchSnapshot();
  });
});
