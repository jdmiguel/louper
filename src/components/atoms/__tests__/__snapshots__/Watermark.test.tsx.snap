// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Watermark /> renders the content properly 1`] = `
<div>
  <svg
    height="230"
    viewBox="0 0 230 230"
    width="230"
  >
    <path
      clip-rule="evenodd"
      d="M160.478 79.1946C167.252 86.512 170.663 95.461 170.663 105.893C170.663 112.667 169.872 118.748 168.339 124.187C166.757 129.576 164.78 133.976 162.456 137.388C160.033 140.799 157.116 143.766 153.655 146.337C150.145 148.908 146.931 150.786 144.014 151.973C141.097 153.16 137.735 154.099 133.928 154.742C130.17 155.384 127.352 155.731 125.374 155.879L119.194 156.027L114.25 156.176L107.378 156.324L100.505 156.176L95.561 156.027L89.3807 155.879C87.4031 155.78 84.5849 155.434 80.8273 154.742C77.0203 154.099 73.7077 153.209 70.7412 151.973C67.8241 150.786 64.6104 148.908 61.1 146.337C57.5896 143.766 54.6726 140.799 52.2499 137.388C49.8767 133.976 47.899 129.576 46.3663 124.187C44.8336 118.748 44.0426 112.667 44.0426 105.893C44.0426 95.461 47.454 86.5614 54.2276 79.1946C53.486 78.8485 53.4365 75.1898 54.1287 68.2185C54.722 61.2472 56.2053 54.8198 58.6279 48.9362C67.0825 49.8261 77.5147 54.622 89.974 63.2249C94.1766 62.1372 99.9613 61.5439 107.328 61.5439C115.041 61.5439 120.826 62.0877 124.682 63.2249C130.368 59.3684 135.807 56.2536 140.998 53.8804C146.239 51.5072 150.046 50.1228 152.469 49.7767L156.028 48.9362C158.451 54.8198 159.934 61.2472 160.528 68.2185C161.269 75.1898 161.22 78.8485 160.478 79.1946ZM107.625 150.255C122.853 150.255 134.324 148.426 142.135 144.767C149.898 141.109 153.853 133.594 153.853 122.172C153.853 115.597 151.381 110.059 146.387 105.659C143.866 103.286 140.85 101.802 137.438 101.259C134.076 100.715 128.885 100.715 121.914 101.259C114.942 101.802 110.196 102.099 107.625 102.099C104.115 102.099 100.357 101.901 95.66 101.555L84.6345 100.863C81.9646 100.764 79.097 101.11 75.9822 101.852C72.8673 102.594 70.2964 103.879 68.2692 105.708C63.5228 109.911 61.1496 115.448 61.1496 122.222C61.1496 133.594 65.0061 141.158 72.719 144.817C80.4319 148.476 91.853 150.305 107.081 150.305L107.625 150.255Z"
      fill="#4d40ba"
      fill-rule="evenodd"
    />
    <path
      d="M135.263 114.988C133.532 112.565 131.505 111.378 129.083 111.378C126.561 111.378 124.386 112.565 122.606 114.988C120.875 117.361 119.985 120.278 119.985 123.788C119.985 127.249 120.875 130.216 122.606 132.589C124.336 134.962 126.512 136.149 129.083 136.149C131.456 136.149 133.483 134.962 135.263 132.589C136.993 130.216 137.883 127.299 137.883 123.788C137.883 120.327 136.993 117.361 135.263 114.988Z"
      fill="#4d40ba"
    />
    <path
      d="M91.3847 115.831C89.6542 113.408 87.6271 112.222 85.2044 112.222C82.6829 112.222 80.5075 113.408 78.7276 115.831C76.9971 118.204 76.1072 121.121 76.1072 124.632C76.1072 128.092 76.9971 131.059 78.7276 133.432C80.458 135.805 82.6335 136.992 85.2044 136.992C87.5777 136.992 89.6048 135.805 91.3847 133.432C93.1152 131.059 94.0051 128.142 94.0051 124.632C94.0051 121.171 93.1152 118.204 91.3847 115.831Z"
      fill="#4d40ba"
    />
    <path
      d="M105.972 205.916C47.5218 205.916 0 159.705 0 102.958C0 46.2104 47.5218 0 105.972 0C164.422 0 211.944 46.2104 211.944 102.958C211.944 159.705 164.422 205.916 105.972 205.916ZM105.972 22.884C60.5199 22.884 23.5539 58.7986 23.5539 102.958C23.5539 147.117 60.5199 183.032 105.972 183.032C151.424 183.032 188.39 147.117 188.39 102.958C188.39 58.7986 151.424 22.884 105.972 22.884Z"
      fill="#4d40ba"
    />
    <path
      d="M162.143 180.946C157.32 176.25 157.33 168.644 162.164 163.959C166.999 159.274 174.827 159.284 179.649 163.98C184.471 168.677 226.39 209.505 226.39 209.505C231.212 214.202 231.202 221.807 226.368 226.492C221.534 231.177 213.706 231.168 208.883 226.471L162.143 180.946Z"
      fill="#4d40ba"
    />
  </svg>
</div>
`;
